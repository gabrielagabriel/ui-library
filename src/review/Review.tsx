import React from 'react'
import cc from 'classcat'

import Paragraph from 'paragraph'
import Caption from 'caption'
import TextTitle from 'typography/title'

export interface ReviewProps {
  readonly className?: string
  readonly isResponse?: boolean
  readonly title: string
  readonly text: string
  readonly formattedDatetime: string
  readonly isoDatetime: string
}

const Review = (props: ReviewProps) => {
  const { className, title, text, formattedDatetime, isoDatetime, isResponse = false } = props

  // For responses, the title is not a rating but the name of the person replying.
  const titleMicroData = isResponse
    ? null
    : {
        itemScope: true,
        itemProp: 'reviewRating',
        itemType: 'https://schema.org/Rating',
      }

  const titleMicroDataContent = isResponse
    ? null
    : {
        itemProp: 'reviewRating',
      }

  return (
    <blockquote
      itemScope
      itemType="http://schema.org/Review"
      className={cc([
        className,
        {
          'kirk-is-review-response': isResponse,
        },
      ])}
    >
      <h2 className="kirk-review-title" {...titleMicroData}>
        <TextTitle {...titleMicroDataContent}>{title}</TextTitle>
      </h2>
      <Paragraph itemProp="reviewBody">{text}</Paragraph>
      <Caption isoDate={isoDatetime}>{formattedDatetime}</Caption>
    </blockquote>
  )
}

export default Review
