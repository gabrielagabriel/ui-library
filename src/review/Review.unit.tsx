import React from 'react'
import { mount } from 'enzyme'

import Review from './index'

const defaultReviewProps = {
  title: 'Review title',
  text: 'Review content',
  formattedDatetime: '05 jul - 17:39',
  isoDatetime: '2017-08-07T14:10:40.526Z',
}

function assertReview(
  review: any,
  expectedTitle: string,
  expectedText: string,
  expectedDate: string,
  isReviewResponse: boolean,
) {
  const title = review.find('h2')
  expect(title.text()).toBe('Review title')
  const paragraph = review.find('p')
  expect(paragraph.text()).toBe('Review content')
  const date = review.find('.kirk-caption')
  expect(date.text()).toBe('05 jul - 17:39')
  const reviewResponse = review.find('.kirk-is-review-response')
  expect(reviewResponse.exists()).toBe(isReviewResponse)
}

it('Should render basic review', () => {
  const review = mount(<Review {...defaultReviewProps} />)
  assertReview(review, 'Review title', 'Review content', '05 jul - 17:39', false)
})

it('Should render review response', () => {
  const review = mount(<Review {...defaultReviewProps} isResponse />)
  assertReview(review, 'Review title', 'Review content', '05 jul - 17:39', true)
})
