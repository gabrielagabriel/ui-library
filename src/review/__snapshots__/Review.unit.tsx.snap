// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render review microdata annotations and HTML 1`] = `
Array [
  .c3 {
  margin: 0;
  font-weight: 400;
}

.c3.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-body,
.c3.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c3.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c3.kirk-text-subheader,
.c3.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-bodyStrong,
.c3.kirk-text-subheaderStrong,
.c3.kirk-text-titleStrong {
  font-weight: 500;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px 0;
}

.c2 .kirk-button {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c4 {
  font-size: 13px;
  line-height: 16px;
  color: #708C91;
}

.c4 .kirk-link {
  font-size: 13px;
  line-height: 16px;
  color: #708C91;
}

.c1 {
  color: #054752;
  font-size: 18px;
  font-weight: 400 line-height:20px;
}

.c0 {
  padding: 8px 0;
  margin: 0;
}

.c0.kirk-is-review-response {
  padding-left: 24px;
}

.c0 .kirk-review-title {
  margin: 0;
  font-size: 1em;
}

<blockquote
    className="c0"
    itemScope={true}
    itemType="http://schema.org/Review"
  >
    <h2
      className="kirk-review-title"
      itemProp="reviewRating"
      itemScope={true}
      itemType="https://schema.org/Rating"
    >
      <span
        className="c1"
        itemProp="reviewRating"
      >
        Review title
      </span>
    </h2>
    <div
      className="c2"
      role="presentation"
    >
      <p
        className="kirk-text kirk-text-body c3"
        itemProp="reviewBody"
      >
        Review content
      </p>
    </div>
    <div
      className="kirk-caption c4"
    >
      <time
        dateTime="2017-08-07T14:10:40.526Z"
      >
        05 jul - 17:39
      </time>
    </div>
  </blockquote>,
  .c3 {
  margin: 0;
  font-weight: 400;
}

.c3.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-body,
.c3.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c3.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c3.kirk-text-subheader,
.c3.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-bodyStrong,
.c3.kirk-text-subheaderStrong,
.c3.kirk-text-titleStrong {
  font-weight: 500;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px 0;
}

.c2 .kirk-button {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c4 {
  font-size: 13px;
  line-height: 16px;
  color: #708C91;
}

.c4 .kirk-link {
  font-size: 13px;
  line-height: 16px;
  color: #708C91;
}

.c1 {
  color: #054752;
  font-size: 18px;
  font-weight: 400 line-height:20px;
}

.c0 {
  padding: 8px 0;
  margin: 0;
}

.c0.kirk-is-review-response {
  padding-left: 24px;
}

.c0 .kirk-review-title {
  margin: 0;
  font-size: 1em;
}

<blockquote
    className="c0 kirk-is-review-response"
    itemScope={true}
    itemType="http://schema.org/Review"
  >
    <h2
      className="kirk-review-title"
    >
      <span
        className="c1"
      >
        Review title
      </span>
    </h2>
    <div
      className="c2"
      role="presentation"
    >
      <p
        className="kirk-text kirk-text-body c3"
        itemProp="reviewBody"
      >
        Review content
      </p>
    </div>
    <div
      className="kirk-caption c4"
    >
      <time
        dateTime="2017-08-07T14:10:40.526Z"
      >
        05 jul - 17:39
      </time>
    </div>
  </blockquote>,
]
`;
